pytorch_lightning/utilities/device_parser.py

```diff
@@ -188,6 +192,21 @@ def _get_all_available_gpus() -> List[int]:
     return list(range(torch.cuda.device_count()))
 
 
+def _check_unique(device_ids: List[int]) -> None:
+    """
+    Checks that the device_ids are unique.
+
+    Args:
+        device_ids: list of ints corresponding to gpus/tpu_cores indices
+
+    Raises:
+        MisconfigurationException:
+            If ``device_ids`` of GPU/TPUs aren't unique
```

Comment:
Does TPUs need this?

cc @kaushikb11

Reply:
No, it doesn't. As TPUs support training on either 1 or 8 cores or on a single core with a particular index such as `[1]`. We handle such parsing logic internally.

Reply:
Thanks for your review. I have updated it.